File: test-plugins/org.yakindu.sct.generator.cpp.test/gtests/LogicalOrTest/LogicalOrTests.cc
#include <string>
#include "gtest/gtest.h"
#include "LogicalOr.h"
#include "sc_runner.h"
#include "sc_types.h"

namespace  {

LogicalOr* statechart;



//! The timers are managed by a timer service. */
static SctUnitRunner * runner;

class LogicalOrTests : public ::testing::Test{
	protected:
	virtual void SetUp() {
		statechart = new LogicalOr();
		statechart->init();
		runner = new SctUnitRunner(
			statechart,
			false,
			200
		);
	}
	virtual void TearDown() {
		delete statechart;
		delete runner;
	}
};


TEST_F(LogicalOrTests, operandEvaluationOrder) {
	
	statechart->enter();
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_x()== 4l && statechart->getDefaultSCI()->get_b()) << "logical or expression was executed in wrong order..." ;
	
	
}

	}

File: test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/LogicalOrTests.java

package org.yakindu.sct.generator.java.test;

import org.junit.*;
import static org.junit.Assert.*;
import org.yakindu.scr.logicalor.LogicalOrStatemachine;
import org.yakindu.scr.logicalor.LogicalOrStatemachine.State;	
import org.yakindu.scr.VirtualTimer;
import org.yakindu.scr.VirtualTimer.VirtualTimeTask;
import org.yakindu.scr.VirtualTimer.CycleTimeEventTask;

/**
 * Unit TestCase for LogicalOr
 */
@SuppressWarnings("all")
public class LogicalOrTests {
	
	private LogicalOrStatemachine statemachine;	
	private VirtualTimer timer;
	
	
	@Before
	public void logicalOrTests_setUp() {
		statemachine = new LogicalOrStatemachine();
		timer = new VirtualTimer(200);
		timer.schedulePeriodicalTask(new CycleTimeEventTask(statemachine), 200, 200);
		
		statemachine.init();
		
	}

	@After
	public void logicalOrTests_tearDown() {
		statemachine = null;
		
		timer = null;
	}
	
	@Test
	public void operandEvaluationOrder() {
		statemachine.enter();
		timer.cycleLeap(1);
		assertTrue("logical or expression was executed in wrong order...",statemachine.getX() == 4l && statemachine.getB());
	}
}
